.gitignore

```diff
@@ -1,6 +1,5 @@
 node_modules
 yarn.lock
-package-lock.json
```

Comment:
Can you help me understand why this has been removed from the `.gitignore`?

Reply:
updates PR text. The GitHub action uses this and errors without a lockfile.
I think it uses it to stay consisten between runs (and for installing with the purpose-built npm ci, instead of npm install).
The warning text I got also told me yarn/pnpm lockfiles were supported too.

Reply:
The first second of this video is quite entertaining: https://www.youtube.com/watch?v=i-AuSqlRRpQ
I really like the explanation, and especially the section on what to do when a lockfile gets merge conflicts.

Anyway. We should keep one, and stick to that as mixing different package managers leads to problems.
I picked npm in this PR as that's the default and comes bundled with node.js, but a change to a different manager is possible too!

Reply:
Hey, sorry for the delay in response.

That video was pretty convincing - I'm on board with committing a lockfile. I definitely agree that if we're going to align to a lockfile, it needs to be npm's `package-lock.json`, since having npm installed is a much better guarantee than having Yarn installed.